src/UniswapV4MulticurveRehypeInitializerHook.sol

```diff
@@ -0,0 +1,687 @@
+            excess0 = fees0 > depositAmount0 ? fees0 - depositAmount0 : 0;
+            excess1 = 0;
+        }
+    }
+
+    function _initializePositionIfNeeded(
+        Position storage position,
+        PoolKey memory key,
+        PoolId poolId
+    ) internal {
+        if (position.salt == bytes32(0)) {
+            position.tickLower = TickMath.minUsableTick(key.tickSpacing);
+            position.tickUpper = TickMath.maxUsableTick(key.tickSpacing);
+            position.salt = _fullRangeSalt(poolId);
+        }
+    }
+
+    function _fullRangeSalt(
+        PoolId poolId
+    ) internal view returns (bytes32) {
+        return keccak256(abi.encodePacked(address(this), PoolId.unwrap(poolId)));
+    }
+
+    function _score(
+        uint256 excess0,
+        uint256 excess1
+    ) internal pure returns (uint256) {
+        return excess0 > excess1 ? excess0 : excess1;
+    }
+
+    function _abs(
+        int256 value
+    ) internal pure returns (uint256) {
+        return value < 0 ? uint256(-value) : uint256(value);
+    }
+
+    function setFeeDistributionForPool(
+        PoolId poolId,
+        address asset,
+        address numeraire,
+        address buybackDst,
+        uint24 customFee,
+        uint256 assetBuybackPercentWad,
+        uint256 numeraireBuybackPercentWad,
+        uint256 beneficiaryPercentWad,
+        uint256 lpPercentWad
+    ) external onlyInitializer(msg.sender) {
+        require(
+            assetBuybackPercentWad + numeraireBuybackPercentWad + beneficiaryPercentWad + lpPercentWad == WAD,
+            FeeDistributionMustAddUpToWAD()
+        );
+        getPoolInfo[poolId] = PoolInfo({ asset: asset, numeraire: numeraire, buybackDst: buybackDst });
+
+        getFeeDistributionInfo[poolId] = FeeDistributionInfo({
+            assetBuybackPercentWad: assetBuybackPercentWad,
+            numeraireBuybackPercentWad: numeraireBuybackPercentWad,
+            beneficiaryPercentWad: beneficiaryPercentWad,
+            lpPercentWad: lpPercentWad
+        });
+
+        getHookFees[poolId] =
+            HookFees({ customFee: customFee, fees0: 0, fees1: 0, beneficiaryFees0: 0, beneficiaryFees1: 0 });
+    }
+
+    function collectFees(
+        PoolId poolId
+    ) external returns (BalanceDelta fees) {
+        HookFees memory hookFees = getHookFees[poolId];
+
+        (,, PoolKey memory poolKey,) =
+            UniswapV4MulticurveRehypeInitializer(INITIALIZER).getState(getPoolInfo[poolId].asset);
+
+        fees = toBalanceDelta(int128(uint128(hookFees.beneficiaryFees0)), int128(uint128(hookFees.beneficiaryFees1)));
+        bool isToken0 = Currency.wrap(getPoolInfo[poolId].asset) == poolKey.currency0;
+
+        Currency.wrap(getPoolInfo[poolId].asset)
+            .transfer(INITIALIZER, isToken0 ? hookFees.beneficiaryFees0 : hookFees.beneficiaryFees1);
+        Currency.wrap(getPoolInfo[poolId].numeraire)
+            .transfer(INITIALIZER, isToken0 ? hookFees.beneficiaryFees1 : hookFees.beneficiaryFees0);
```

Comment:
Would recommend validating that the amount to send is non-zero as some tokens will revert on zero value transfers, which will block the other token from getting transferred